src/main/java/com/iota/iri/BundleValidator.java

```diff
@@ -8,8 +8,40 @@
 
 import java.util.*;
 
+/**
+ * Validates bundles.
+ * <p>
+ * Bundles are a list of transactions that represent an atomic transfer, meaning that either all
+ * transactions inside the bundle will be accepted by the network, or none.
+ *</p>
+ */
 public class BundleValidator {
 
+    /**
+     * Fetches a bundle of transactions identified by the {@code tailHash} and validates them.
+     * <p>
+     * The fetched transactions have the same bundle hash as the transaction identified by {@code tailHash}
+     * The validation ascertains that:
+     * <ol>
+     *     <li>{@code tailHash} has an index of 0</li>
+     *     <li>{@code tailHash} was not already invalidated by a previous call to this method</li>
+     *     <li>That the transactions reference order is consistent with the indexes</li>
```

Comment:
that twice (see title)
```suggestion
     *     <li>The transactions' reference order is consistent with the indexes</li>
```

Reply:
transactions' reference order - maybe be explicit: traversing the trunks?